osf/models/node.py

```diff
@@ -1140,8 +1140,14 @@ def visible_contributors(self):
 
     def get_spam_fields(self, saved_fields):
         # Override for SpamOverrideMixin
-        return self.SPAM_CHECK_FIELDS if self.is_public and 'is_public' in saved_fields else self.SPAM_CHECK_FIELDS.intersection(
-            saved_fields)
+        if self.is_public and 'is_public' in saved_fields:
+            if self.get_addon('forward'):
+                return self.SPAM_CHECK_FIELDS
+            else:
+                return self.SPAM_CHECK_FIELDS.difference('addons_forward_node_settings__url')
+
+        else:
+            return self.SPAM_CHECK_FIELDS.intersection(saved_fields)
```

Comment:
Difference accepts a `set`.
```
check_fields = self.SPAM_CHECK_FIELDS.copy()
if not self.has_addon('forward'):
    check_fields.remove('addons_forward_node_settings__url')
return fields if self.is_public and 'is_public' in saved_fields else check_fields.intersection(saved_fields)
```

Reply:
> Difference accepts a set.

It also accepts a string!  
```
In [215]: {'a', 'b', 'c'}.difference('a').copy()
Out[215]: {'b', 'c'}
In [215]: {'a', 'b', 'c'}.difference('a').copy()
Out[215]: {'b', 'c'}

It's the copy method I don't understand here, but I put in the new commit.

Reply:
It does accept a string, because strings can be reduced to a `set` of characters.
```
In [8]: Node.SPAM_CHECK_FIELDS.difference('addons_forward_node_settings__url')
Out[8]: {'addons_forward_node_settings__url', 'description', 'title'}

In [9]: set('addons_forward_node_settings__url')
Out[9]: {'_', 'a', 'd', 'e', 'f', 'g', 'i', 'l', 'n', 'o', 'r', 's', 't', 'u', 'w'}

In [10]: set({'addons_forward_node_settings__url'})
Out[10]: {'addons_forward_node_settings__url'}

In [11]: Node.SPAM_CHECK_FIELDS.difference({'addons_forward_node_settings__url'})
Out[11]: {'description', 'title'}
```

Reply:
>It's the copy method I don't understand here, but I put in the new commit.

Shallow `copy` of the `SPAM_CHECK_FIELDS` set, otherwise this happens:
```
In [12]: fields = Node.SPAM_CHECK_FIELDS

In [13]: Node.SPAM_CHECK_FIELDS
Out[13]: {'addons_forward_node_settings__url', 'description', 'title'}

In [14]: fields
Out[14]: {'addons_forward_node_settings__url', 'description', 'title'}

In [15]: fields.remove('addons_forward_node_settings__url')

In [16]: fields
Out[16]: {'description', 'title'}

In [17]: Node.SPAM_CHECK_FIELDS
Out[17]: {'description', 'title'}
```

Reply:
That's what I get for testing with a single character.